### PR TITLE
Remove save_state_warning in LambdaLR

### LINKED ISSUE #46405: Allow downstream users to suppress Save Optimizer warnings
## 🚀 Feature
Enable downstream users of this library to suppress lr_scheduler save_state_warning. (Propose to add an argument to LambdaLR [torch/optim/lr_scheduler.py])

If set to true, the warnings.warn(SAVE_STATE_WARNING, user_warning) that prints "Please also save or load the state of the optimizer when saving or loading the scheduler."  will not be generated.

## Motivation
Various bugs / discussions exist because users of various libraries are confused by this warning. Huggingface recently pushed a change to catch and suppress this warning.

[Recent Pytorch discussion](https://discuss.pytorch.org/t/weird-warning-popping-out-regarding-learning-rate-scheduler/99293)

[Huggingface issue](https://github.com/huggingface/transformers/issues/7765)

[Huggingface solution to deal with "the annoying warning"](https://github.com/huggingface/transformers/pull/7401)

## Pitch
Propose to add an argument to LambdaLR [torch/optim/lr_scheduler.py](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py)
* state_dict(..., suppress_state_warning=False)
* load_state_dict(..., suppress_state_warning=False)

Default false preserves the warning for everyone, except those who explicitly choose to set the flag, presumably because they have appropriately saved the optimizer.

## Alternatives
Huggingface implemented a wrapper to catch and suppress the warning but this is fragile.

## Concerns
Maybe there's some plumbing that should be updated to use this new flag, but once we provide the option to use the flag, others can begin implementing on their own.

cc @vincentqb

### PR DESCRIPTION
**Pull request by jsrozner**
Fixes #46405, #43352 

I updated the docstring in the local file (function level comments). Do I also need to edit somewhere else or recompile docstrings?

Also, though I didn't change any types here, how is typing (for IDE type checking) documentation generated / used)?

### CODE REVIEW

**Comment by vincentqb** [comment]
> I updated the docstring in the local file (function level comments). Do I also need to edit somewhere else or recompile docstrings?

As mentioned in [comment](https://github.com/pytorch/pytorch/issues/46405#issuecomment-713979091), the docstring will automatically regenerate the doc on master. :)

> Also, though I didn't change any types here, how is typing (for IDE type checking) documentation generated / used)?

Since this is just updating the warning and the documentation, there is no need update the typing.

**Comment by jsrozner** [comment]
> > Also, though I didn't change any types here, how is typing (for IDE type checking) documentation generated / used)?
> 
> Since this is just updating the warning and the documentation, there is no need update the typing.

Yes, agreed. But I was curious how the pytorch library handles typing since it is not inlined in, e.g., these function definitions

Is anything else needed by me to finish this pull request?

**Review by vincentqb — APPROVED** [review]
LGTM :)

**Comment by jsrozner** [comment]
@vincentqb 
Can we merge?

**Comment by stas00** [comment]
why was it removed?

That broke transformers where it's used:
https://github.com/huggingface/transformers/blob/aa0969c2f3cf5cf7ce5690e7b77f95b1622988d6/src/transformers/trainer_pt_utils.py#L37
